src/seedu/addressbook/data/person/Name.java

```diff
@@ -43,6 +46,48 @@ public static boolean isValidName(String test) {
     public List<String> getWordsInName() {
         return Arrays.asList(fullName.split("\\s+"));
     }
+    
+    /**
+     * Returns true of the other name is very similar to this name. 
+     * Two names are considered similar if they have at least 
+     * 50 percent match in case-insensitive words, in any order.
+     */
+    public boolean isSimilar(Name other) {
+        if (other == null)
+            return false;
+        
+        List<String> myWords = new ArrayList<String>(getWordsInName());
+        List<String> otherWords = new ArrayList<String>(other.getWordsInName());
+        
+        // make both lists of words into lowercase
+        ListIterator<String> iterator = myWords.listIterator();
+        while (iterator.hasNext())
```

Comment:
coding standard violation :-1:  the brace should be immediately after `while` statement not next line